packages/base64/base64_test.js

```diff
@@ -10,50 +12,45 @@ var asciiToArray = function (str) {
   return arr;
 };
 
-var arrayToAscii = function (arr) {
-  var res = [];
-  for (var i = 0; i < arr.length; i++) {
-    res.push(String.fromCharCode(arr[i]));
-  }
-  return res.join("");
-};
+const arrayToAscii = arr =>
+  arr.reduce(
+    (prev, charCode) => [...prev, String.fromCharCode(charCode)], []
```

Comment:
Not the biggest deal since this is in tests, so I guess this is more me expressing preference: I personally don't find this notation easier to understand than the original, and in my experience this would be much less performant than the previous implementation (the `for` loop), especially since a new `Array` is being initialized `[ ... ]` and having its values copied into it (via spread) on each iteration.

A similar notation, which should perform significantly better, would be `push`-ing onto the Array and returning it, which would maintain and manipulate the reference to the previous object, rather than allocating a a new one:

```js
const arrayToAscii = arr => arr
  .reduce((prev, charCode) =>
    prev.push(String.fromCharCode(charCode)) && prev, [])
  .join('');
```

Reply:
Good point!